src/Database/Type/DateType.php

```diff
@@ -102,14 +103,61 @@ public function useMutable()
      */
     public function marshal($value): ?DateTimeInterface
     {
-        $date = parent::marshal($value);
-        /** @psalm-var \DateTime|\DateTimeImmutable|null $date */
-        if ($date && !$date instanceof I18nDateTimeInterface) {
-            // Clear time manually when I18n types aren't available and raw DateTime used
-            $date = $date->setTime(0, 0, 0);
+        if ($value instanceof DateTimeInterface) {
+            return new FrozenDate('@' . $value->getTimestamp());
```

Comment:
Why do you need to use the timestamp here? Can you just pass the DateTimeInterface in?

Reply:
PHPstan was complaining about parameter types. This seemed liked a simple way to solve that.

Reply:
Ok. We don't need it to be perfect for 4.5 then.